plugin/forward/setup.go

```diff
@@ -204,7 +206,11 @@ func parseBlock(c *caddy.Controller, f *Forward) error {
 				if !c.NextArg() {
 					return c.ArgErr()
 				}
-				f.opts.hcDomain = c.Val()
+				hcDomain := c.Val()
+				if _, ok := dns.IsDomainName(hcDomain); !ok {
```

Comment:
You can normalize here, That way you don’t have to do it later.

Reply:
Ok, no problem, thanks